src/Microsoft.Azure.WebJobs.Host/CompositeTraceWriter.cs

```diff
@@ -25,7 +25,10 @@ public CompositeTraceWriter(IEnumerable<TraceWriter> traceWriters, TextWriter te
             var traceWritersCopy = traceWriters?.ToList() ?? new List<TraceWriter>();
             _innerTraceWriters = traceWritersCopy.AsReadOnly();
 
-            _innerTextWriter = textWriter;
+            if (textWriter != null)
+            {
+                _innerTextWriter = TextWriter.Synchronized(textWriter);
```

Comment:
The other innerTraceWriters we're delegating to as well are already synchronized?

Reply:
The other few places we use a TextWriter, we're either wrapping it in Synchronized already (I fixed a few a while back) or they're not being shared.